From 29da1ea82810214f5d4188111a61aa6dac048ffe Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Konrad=20Zapa=C5=82owicz?=
 <konrad.zapalowicz@chainguard.dev>
Date: Fri, 5 Sep 2025 11:18:21 +0200
Subject: [PATCH 2/4] refactor(controller): add Chainguart to supported OSes
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The flux-operator performs a set of preflight checks including making sure it
is running on a supported OS. If that is not the case it errors and exits the
reconcilation loop.

This commit extends the set of supported OSes with Chainguard so that
flux-operator can be run with Wolfi as a base.

Signed-off-by: Konrad Zapałowicz <konrad.zapalowicz@chainguard.dev>
---
 internal/controller/fluxinstance_controller.go | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/internal/controller/fluxinstance_controller.go b/internal/controller/fluxinstance_controller.go
index baaa8dc..ba7d727 100644
--- a/internal/controller/fluxinstance_controller.go
+++ b/internal/controller/fluxinstance_controller.go
@@ -158,6 +158,11 @@ func (r *FluxInstanceReconciler) reconcile(ctx context.Context,
 		builder.WithMinVersion("2.2.0"),
 		builder.WithContainerOS("distroless", 12),
 		builder.WithContainerOS("rhel", 8),
+		// The OS differs depending on the audience: for public images that is
+		// Wolfi and for commercial offering it is Chainguard. The flux-operator is
+		// not a public image hence Chainguard here. See:
+		// https://chainguard-dev.slack.com/archives/C04PYHWPE1F/p1756218379927789
+		builder.WithContainerOS("Chainguard", 20230214),
 	); err != nil {
 		// If this happens, then the operator image has been tampered with
 		// e.g.: the manifests are missing, or the Flux / OS version is not supported.
-- 
2.50.1

